### PR TITLE
Add support for open source model inference 

### PR DESCRIPTION
**Pull request by libowen2121**
This PR introduces support for open source model inference, using an OpenAI-compatible interface. We have currently integrated `deepseek-coder-33b-instruct` and `deepseek-coder-6.7b-instruct`.

### CODE REVIEW

File: inference/run_api.py [hunk 1]
```diff
@@ -42,6 +44,8 @@
     "gpt-4-0613": 8_192,
     "gpt-4-1106-preview": 128_000,
     "gpt-4-0125-preview": 128_000,
+    "deepseek-coder-6.7b-instruct": 32_000,
+    "deepseek-coder-33b-instruct": 32_000,
```
Comment (huybery) [inline inference/run_api.py:48]: why 32k?

File: inference/run_api.py [hunk 2]
```diff
@@ -165,6 +205,16 @@ def gpt_tokenize(string: str, encoding) -> int:
...
 
+def get_model_id(model_name: str) -> str:
+    """Returns normalized model id from abitrary serving model name.
+    E.g., abitrary_path/deepseek-coder-33b -> deepseek.
+    """
+    if "deepseek" in model_name:
+        return "deepseek"
+    if "codellama" in model_name:
+        return "codelllma"
+
```
Comment (huybery) [inline inference/run_api.py:218]: What happens if none of the conditions are triggered?

**Review by huybery — CHANGES_REQUESTED** [review]

File: inference/make_datasets/tokenize_dataset.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 import tiktoken
 from datasets import disable_caching, load_from_disk, load_dataset
 from tqdm.auto import tqdm
-from transformers import LlamaTokenizer
+from transformers import LlamaTokenizer, AutoTokenizer
```
Comment (JustinLin610) [inline inference/make_datasets/tokenize_dataset.py:14]: I think here we only need AutoTokenizer?

File: inference/run_api.py (hunk 2 above)
Comment (JustinLin610) [reply inference/run_api.py:218]: There should be a value error in the else condition

File: inference/run_api.py (hunk 1 above)
Comment (libowen2121) [reply inference/run_api.py:48]: Will make it configurable later.

File: inference/run_api.py (hunk 2 above)
Comment (libowen2121) [reply inference/run_api.py:218]: Have resolved it.

**Review by JustinLin610 — APPROVED** [review]

**Pull request merged**